index.js

```diff
@@ -1,30 +1,55 @@
 var fs = require("fs");
 var path = require("path");
-var detective = require('detective');
+var Walker = require('node-source-walk');
 var q = require('q');
 var walkdir = require("walkdir");
 var _ = require('lodash');
 var minimatch = require('minimatch');
 var util = require('util');
 
+function getArgumentFromCall(node) {
+  return node.type === 'CallExpression' && node.arguments[0]
+    ? node.arguments[0].value
+    : undefined;
+}
+
+function isRequireFunction(node) {
+  var callee = node.callee;
+  return callee && callee.type === 'Identifier' && callee.name === 'require' &&
+    getArgumentFromCall(node);
+}
+
+function isGruntLoadTaskCall(node) {
+  var callee = node.callee;
+  return callee && callee.property && callee.property.name === 'loadNpmTasks' &&
+    getArgumentFromCall(node);
+}
+
+function isImportDeclaration(node) {
+  return node.type === 'ImportDeclaration' && node.source && node.source.value;
+}
+
 function getModulesRequiredFromFilename(filename) {
   var content = fs.readFileSync(filename, "utf-8");
+  if (!content) {
+    return [];
+  }
+
+  var walker = new Walker();
```

Comment:
Another idea is to pass the acorn object as depcheck argument. I implemented JSX support in [my local repo](https://github.com/rumpl/depcheck/compare/bb83d7c420804f19062c70f8ddc542235f966705...lijunle:jsx-support), it is using [acorn-jsx](https://www.npmjs.com/package/acorn-jsx) under the hood.